ConvertNER/convert_NER.py

```diff
@@ -0,0 +1,212 @@
+            return c
+
+def get_orth(seg):
+    morph = get(seg, 'type', 'morph')
+    orth = get(morph, 'name', 'orth')
+    return orth[0].text if orth is not None else None
+
+def get_named(seg):
+    named = get(seg, 'type', 'named')
+    orth = get(named, 'name', 'orth')
+    return orth[0].text if orth is not None else None
+
+def get_named_type(seg):
+    named = get(seg, 'type', 'named')
+    type = get(named, 'name', 'type')
+    return type[0].attrib['value']
+
+def get_ctag(seg):
+    morph = get(seg, 'type', 'morph')
+    interps = get(morph, 'name', 'interps')
+    lex = get(interps, 'type', 'lex')
+    ctag = get(lex, 'name', 'ctag')
+    return ctag[0].attrib['value']
+
+def get_corresp_morph(sent):
+    return sent.attrib['corresp'].split('#')[1]
+
+def get_entity_maps(root):
+    result = {}
+    for sent in root.iter('{http://www.tei-c.org/ns/1.0}s'):
+        tmp = []
+        for seg in sent:
+            text = get_named(seg)
+            type = get_named_type(seg)
+            tmp += [(text, type)]
+
+        result[get_corresp_morph(sent)] = dict(tmp)
+
+    return result
+
+def get_segmentation_text_maps(root):
+    res = {}
+    for paragraph in root.iter('{http://www.tei-c.org/ns/1.0}p'):
+        key = paragraph.attrib['{http://www.w3.org/XML/1998/namespace}id']
+        value = paragraph.attrib['corresp'].split('#')[1]
+        res[key]=value
+
+    return res
+
+def get_text_maps(root):
+    result = {}
+    for paragraph in root.iter('{http://www.tei-c.org/ns/1.0}div'):
+        key = paragraph.attrib['{http://www.w3.org/XML/1998/namespace}id']
+        text = ''
+        for child in paragraph:
+            text += child.text
+
+        result[key]=text
+
+    return result
+
+def get_sent_id(sent):
+    return sent.attrib['{http://www.w3.org/XML/1998/namespace}id']
+
+def get_paragraph_text(paragraph, segm_text_map, text_maps):
+    paragraph_id = paragraph.attrib['corresp'].split('#')[1]
+    return text_maps[segm_text_map[paragraph_id]]
+
+def set_biluo_tags(sentences, tags):
+    i = 0
+    for sent in sentences:
+        for token in sent:
+            token['ner'] = tags[i]
+            i += 1
+
+    return sentences
+
+def required_files_exist(dir):
+    required_files = [segmentation_xml, text_xml, named_xml, morphosyntax_xml]
+    for file in required_files:
+        if not os.path.isfile(os.path.join(path_prefix,corpus_path,dir,file)):
+            return False
+
+    return True
+
+nlp = spacy.load('en_core_web_sm')
+doc_id = 0
+corpus = []
+
+NE_njkp_to_spacy = {'persName': 'PERSON',
+ 'placeName': 'LOC',
+ 'orgName': 'ORG',
+ 'date': 'DATE',
+ 'time': 'TIME',
+ 'geogName': 'LOC'}
+
+for f in os.listdir(os.path.join(path_prefix, corpus_path)):
+    doc_json = {}
+    current_folder = f
+
+    if not os.path.isdir((os.path.join(path_prefix,corpus_path,current_folder))):
+        continue
+
+    if not required_files_exist(current_folder):
+        # doc_id +=1 ?
+        continue
```

Comment:
Why ifs when we do not handle them?

Reply:
We skip the docs that don't have the required annotations (certain .xml files) - as far as I remember not all of them do